addon/components/polaris-color-picker/slidable.js

```diff
@@ -142,19 +110,32 @@ export default Component.extend({
     }
 
     const rect = getRectForNode(element);
+
     moveHandler({
       x: clientX - rect.left,
       y: clientY - rect.top,
     });
-  },
+  }
+
+  @computed('contentId')
+  get element() {
+    return document.querySelector(`#${this.contentId}`);
+  }
+
+  @action
+  setContentId(element, value) {
```

Comment:
is this copied from somewhere? I don't understand where it's coming from, can't see it in the version prior to these changes

Reply:
It's slighty different than the old version, it used to be

`setContentId(element, [value])`

because it used to get called from the `on-insert` modifier, but now it gets called from within `setContentIdAndTriggerHeightChanged` below.

Reply:
would you mind linking me to the old version here on github?

Reply:
It's here in my first commit - this link doesn't seem to scroll to the line changes, but its `slidable.js`, lines 150-153 - https://github.com/smile-io/ember-polaris/pull/493/commits/0c66046427b0f0b5bf6dfe02813057b85697560b#diff-160b5cb0d5fd7505911f238ca1295ee4R150-R153

Reply:
kk....do you think this is needed? why are we adding an ID attirbute to the new wrapping `div` in the template that was added after making this component tag-less?

Reply:
The component has other methods like `handleDraggerMove` which need to grab the main element, so we need some way of grabbing it. I left the `element` CP in here as [mentioned](https://github.com/smile-io/ember-polaris/pull/493#discussion_r375989667) earlier so other methods could `.get` the element when needed.

I'm open to doing it differently but we do need some way to access the main element.

Reply:
this should be much simpler....please check [ember-render-modifiers](https://github.com/emberjs/ember-render-modifiers) to understand it's usage....it's super simple: `did-insert` will invoke your action with one param `element` (the element on which `did-insert` is added) and whatever else you're passing to it.....just set this `element` argument you receive in your action and set it on this component

there's no need `guidFor` and appending `content`....etc (which is why I was confused)

Reply:
Ahh I did that originally, but got the error

`Cannot set property element of [object Object] which has only a getter`

I thought maybe it was something weird with the modifiers but it turns out it just doesn't want me using the key name `"element"`.. 🤦‍♂ So yeah I can just do that.

Reply:
welp, `element` is a reserved word in ember so makes sense

Reply:
Yeah, strange that it let me overwrite it with a CP tho.

Reply:
> Yeah, strange that it let me overwrite it with a CP tho.

I guess overwriting replaces the entire setter/getter combo, while setting tries to access a non-existent setter on the existing `element` property 🤷‍♂